### PR TITLE
Migrate pallet-election-provider-support-benchmarking benchmark to v2

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Part of:

- #6202.

### CODE REVIEW

File: substrate/frame/election-provider-support/benchmarking/src/inner.rs [hunk 1]
```diff
@@ -20,17 +20,19 @@
...
 
 const VOTERS: [u32; 2] = [1_000, 2_000];
 const TARGETS: [u32; 2] = [500, 1_000];
 const VOTES_PER_VOTER: [u32; 2] = [5, 16];
-
 const SEED: u32 = 999;
+
+pub trait Config: frame_system::Config {}
+
+pub struct Pallet<T: Config>(frame_system::Pallet<T>);
```
Comment (gui1117) [inline substrate/frame/election-provider-support/benchmarking/src/inner.rs:34]: why are they necessary?

Comment (AurevoirXavier) [reply substrate/frame/election-provider-support/benchmarking/src/inner.rs:34]: These come from the old file.

Comment (gui1117) [reply substrate/frame/election-provider-support/benchmarking/src/inner.rs:34]: Ah my bad, I didn't see the were written above before.

File: prdoc/pr_6315.prdoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+title: Migrate pallet-election-provider-support-benchmarking benchmark to v2
+doc:
+- audience: Runtime Dev
+  description: |-
+    Part of:
+
+    - #6202.
```
Comment (gui1117) [inline prdoc/pr_6315.prdoc:7]:
```suggestion
  description: |-
    Migrate pallet-election-provider-support-benchmarking benchmark to v2
```

**Review by gui1117 — APPROVED** [review]

**Review by re-gius — APPROVED** [review]

**Pull request merged**